packages/airnode-protocol/contracts/rrp/AirnodeRrp.sol

```diff
@@ -253,7 +253,13 @@ contract AirnodeRrp is
             ) == requestIdToFulfillmentParameters[requestId],
             "Invalid request fulfillment"
         );
-        verifySignature(requestId, airnode, data, signature);
+        require(
+            (
+                keccak256(abi.encodePacked(requestId, data))
+                    .toEthSignedMessageHash()
+            ).recover(signature) == airnode,
+            "Signature mismatch"
+        );
```

Comment:
Shouldn't we have an internal pure `verifyHash(...)` function that gets called here and in `verifyData()` (329-335)

Reply:
Done here 27ddc8907cdde73fab51201727c25afb5abc48b1